### ISSUE TITLE
Table names not handled consistently

### ISSUE THREAD

**Issue by iandunn** [WordPress/wporg-code-analysis#42]
```php
// Fails
$wpdb->query( sprintf( "TRUNCATE TABLE %s;", self::get_table_name() ) );

// Passes
$table_name = self::get_table_name();
$wpdb->query( "TRUNCATE TABLE $table_name;" );
```

Since [there isn't a best practice for escaping table names yet](https://core.trac.wordpress.org/ticket/52506), should we just always allow them to be unescaped?

**Comment by iandunn** [comment]
🤔 , it looks like anything assigned to a variable is allowed, even if it's obviously unsafe:

```php
// Passes
$table_name = $_GET['table'];
$wpdb->query( "TRUNCATE TABLE $table_name;" );
```

Is that intentional, to avoid false positives or something? If so, then maybe that's the actual problem to solve?

**Comment by iandunn** [comment]
🤔 , this fails, though, even though it's assigned to a var. 

```php
// Fails
$reimbursements_index = Reimbursements_Dashboard\get_index_table_name();

$paid_reimbursements = $wpdb->get_results( "
	SELECT blog_id, request_id, date_paid
	FROM `$reimbursements_index`
	WHERE status = 'wcb-paid'
" );
```

**Comment by iandunn** [comment]
Ah, it's because `$table_name` is safelisted:

https://github.com/WordPress/wporg-code-analysis/blob/10532dcc061f51bcc3a429d0649d182d7096125d/MinimalPluginStandard/Sniffs/DirectDBSniff.php#L126-L132
